docs/import/list_ee_assets.js

```diff
@@ -1,41 +1,117 @@
 import {eeLegacyPathPrefix, legacyStateDir} from '../ee_paths.js';
 import {LayerType} from '../firebase_layers.js';
+import {convertEeObjectToPromise} from '../map_util.js';
 
 export {getDisasterAssetsFromEe, getStatesAssetsFromEe};
 
 /**
- * Requests all assets in ee directories corresponding to given states.
+ * Cache for the results of listEeAssets for each state.
+ * @type {Map<string, Promise<Map<string, string>>>} Promise with a map whose
+ *     keys are the asset names, and values are types
+ */
+const stateAssetPromises = new Map();
+
+/**
+ * Gets all assets in ee directories corresponding to given states. Caches
+ * results of calls to listEeAssets for each state. Marks assets with their type
+ * and whether or not they should be disabled when put into a select. Here,
+ * disabling any assets that aren't feature collections.
  * @param {Array<string>} states e.g. ['WA']
- * @return {Promise<Array<Array<string | Array<string>>>>} 2-d array of all
- *     retrieved assets in the form [['WA', {'asset/path': LayerType,...}], ...]
+ * @return {Promise<Array<Array<string, {disable: boolean}>>>} 2-d array of all
+ *     retrieved assets in the form [['WA', {'path': {disable: true}}],...]
```

Comment:
Are you anticipating more attributes to add to each asset? If not, why not just a boolean, instead of a dictionary?

Reply:
Partial response to your partial review (: I thought since the disaster one has a full object because it needs the type as well the symmetry was nice. And I can see it being helpful if we ever want to expose the type. But can remove if you'd prefer.

Reply:
I dunno: we're silently filtering out folders, so it's not like we show everything blindly. feature collections without geometries are a harder case for the user to understand, knowing that you can't use an image as the SVI asset without it being explicitly disabled seems more like an expectation we could have. You can decide!